src/ensembl/src/content/app/entity-viewer/state/sidebar/entityViewerSidebarActions.ts

```diff
@@ -0,0 +1,52 @@
+import { createAction } from 'typesafe-actions';
+import { ActionCreator, Action } from 'redux';
+import { ThunkAction } from 'redux-thunk';
+
+import { getEntityViewerActiveGenomeId } from '../general/entityViewerGeneralSelectors';
+import { isEntityViewerSidebarOpen } from 'src/content/app/entity-viewer/state/sidebar/entityViewerSidebarSelectors';
+
+import {
+  EntityViewerSidebarStateForGenome,
+  SidebarTabName
+} from './entityViewerSidebarState';
+import { RootState } from 'src/store';
+
+export const updateSidebar = createAction(
+  'entity-viewer-sidebar/update-sidebar'
+)<{ genomeId: string; fragment: Partial<EntityViewerSidebarStateForGenome> }>();
+
+export const setSidebarTabName: ActionCreator<ThunkAction<
+  void,
+  any,
+  null,
+  Action<string>
+>> = (tabName: SidebarTabName) => (dispatch, getState: () => RootState) => {
+  const activeGenomeId = getEntityViewerActiveGenomeId(getState());
+  if (!activeGenomeId) {
+    return;
+  }
+  dispatch(
+    updateSidebar({
+      genomeId: activeGenomeId,
+      fragment: { activeTabName: tabName }
+    })
+  );
+};
+
+export const toggleSidebar: ActionCreator<ThunkAction<
+  void,
+  any,
+  null,
+  Action<string>
+>> = (isOpen?: boolean) => (dispatch, getState: () => RootState) => {
+  const state = getState();
+  const genomeId = getEntityViewerActiveGenomeId(state);
+  if (!genomeId) {
+    return;
+  }
+  if (isOpen === undefined) {
+    const isCurrentlyOpen = isEntityViewerSidebarOpen(state);
+    isOpen = !isCurrentlyOpen;
```

Comment:
This is quite confusing to read. Can `isOpen` be renamed to something like `shouldBeOpen` ?

Reply:
`should` is also confusing (if `shouldBeOpen` is false, it does not mean that the thing should be closed).
We can change it to an enum value:

```
enum SidebarPosition = {
  OPEN = 'open',
  CLOSED = 'closed'
};
```
and pass that instead of a boolean. It might be a bit of an overkill though.

Reply:
If `shouldBeOpen` is false, would it not be easy to consider it as "should not be open"?

Reply:
I pushed a commit that uses the Status enum for expressing the open or closed state of the sidebar:
https://github.com/Ensembl/ensembl-client/pull/237/commits/56240d9d14858ca5ef2bf0d1da836749936df309

Reply:
Cool, I guess you missed my previous comment?

Reply:
No I saw it, but if you think that `isOpen` is confusing, then `shouldBeOpen` will probably be just as confusing.